modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators.Tests/MustBeVariantAnalyzerTests.cs

```diff
@@ -0,0 +1,33 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+
+namespace Godot.SourceGenerators.Tests;
+
+public class MustBeVariantAnalyzerTests
+{
+    [Fact]
+    public async void GenericTypeArgumentMustBeVariantTest()
+    {
+        const string MustBeVariantGD0301 = "MustBeVariant.GD0301.cs";
+        await CSharpAnalyzerVerifier<MustBeVariantAnalyzer>.Verify(MustBeVariantGD0301,
+            new DiagnosticResult("GD0301", DiagnosticSeverity.Error)
+                .WithSpan(MustBeVariantGD0301, 11, 16, 11, 22)
+                .WithArguments(MustBeVariantGD0301)
```

Comment:
I squashed the commits but I deleted the original branch in the process, as you can see in the history log... A new branch with exactly the same name and one single commit (with the `{|GD0301:...|}` change requested) has been created. Not sure if this PR could be reopened again or I should create a new one, just let me know. Sorry for the mess! :)

Reply:
> I squashed the commits but I deleted the original branch in the process, as you can see in the history log... A new branch with exactly the same name and one single commit (with the `{|GD0301:...|}` change requested) has been created. Not sure if this PR could be reopened again or I should create a new one [from the same branch again](https://github.com/godotengine/godot/compare/master...avilches:godot:must-be-variant-tests), just let me know. Sorry for the mess! :)

Hi @AThousandShips should I create a new PR? or could you please reopen this one with the same branch, which is already restored?

Reply:
I can't reopen this as the branch is deleted, maybe you can push to it again but otherwise I can't do anything here 🙂

I think a deleted branch can't be reopened in the system

Reply:
I created a new one. Sorry for the mess: https://github.com/godotengine/godot/pull/86528